daemons/ipa-kdb/ipa_kdb_principals.c

```diff
@@ -1057,7 +1059,8 @@ krb5_error_code ipadb_find_principal(krb5_context kcontext,
         for (int i = 0; vals[i]; i++) {
 #ifdef KRB5_KDB_FLAG_ALIAS_OK
             if ((flags & KRB5_KDB_FLAG_ALIAS_OK) == 0) {
-                found = strcmp(vals[i]->bv_val, *principal) == 0;
+                found = ((vals[i]->bv_len == princ_len) &&
+                         strncmp(vals[i]->bv_val, *principal, vals[i]->bv_len) == 0);
```

Comment:
Is ``strncmp`` safe? How about a custom function that first compares length of BerValue with strlen(char*), then performs memcmp() iff both are the same size.

Reply:
it is a string but not '\0'-terminated. The principal cannot contain any NUL byte.

Reply:
`strncmp` is fine, this is what krb5 itself uses.

Reply:
Thanks!